tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ContributionTask.java

```diff
@@ -0,0 +1,500 @@
+import org.gradle.api.DefaultTask;
+import org.gradle.api.file.FileSystemOperations;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.process.ExecOperations;
+
+import javax.inject.Inject;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.*;
+
+public abstract class ContributionTask extends DefaultTask {
+    private static final String METADATA_INDEX = "metadata/index.json";
+    private static final String BUILD_FILE = "build.gradle";
+    private static final String USER_CODE_FILTER_FILE = "user-code-filter.json";
+    private static final String REQUIRED_DOCKER_IMAGES_FILE = "required-docker-images.txt";
+
+    @Inject
+    protected abstract ExecOperations getExecOperations();
+
+    @Inject
+    protected abstract FileSystemOperations getFileSystemOperations();
+
+    private Path gradlew;
+
+    private final ObjectMapper objectMapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT).setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+    private Path testsDirectory;
+    private Path metadataDirectory;
+
+    private Coordinates coordinates;
+
+    private record ContributingQuestion(String question, String help) {}
+    private final Map<String, ContributingQuestion> questions = new HashMap<>();
+
+    private void initializeWorkingDirectories(){
+        testsDirectory = Path.of(getProject().file(CoordinateUtils.replace("tests/src/$group$/$artifact$/$version$", coordinates)).getAbsolutePath());
+        metadataDirectory = Path.of(getProject().file(CoordinateUtils.replace("metadata/$group$/$artifact$/$version$", coordinates)).getAbsolutePath());
+        this.gradlew = Path.of(getProject().file("gradlew").getAbsolutePath());
+    }
+
+    private void loadQuestions() throws IOException {
+        File questionsJson = getProject().file("tests/tck-build-logic/src/main/resources/contributing/questions.json");
+        List<Question> contributingQuestions = objectMapper.readValue(questionsJson, new TypeReference<>() {});
+        for (var question : contributingQuestions) {
+            this.questions.put(question.questionKey(), new ContributingQuestion(question.question(), question.help()));
+        }
+    }
+
+    @TaskAction
+    void run() throws IOException {
+        InteractiveTaskUtils.printUserInfo("Hello! This task will help you contributing to metadata repository." +
+                " Please answer the following questions. In case you don't know the answer on the question, type \"help\" for more information");
+
+        loadQuestions();
+
+        this.coordinates = getCoordinates();
+        InteractiveTaskUtils.closeSection();
+
+        Path coordinatesMetadataRoot = getProject().file(CoordinateUtils.replace("metadata/$group$/$artifact$", coordinates)).toPath();
```

Comment:
How do you decide when to use
```
Path.of(<some file>).getAbsolutePath())
```
or
```
<some file>.toPath()
```
?

Reply:
You are right, I should always use `<some file>.toPath()`.